### PR TITLE
Docs: add examples to `istril`/`istriu` docstrings

### PR DESCRIPTION
**Pull request by jishnub**
Also, simplify some of the examples to focus on the structure of the matrices

### CODE REVIEW

File: stdlib/LinearAlgebra/src/generic.jl [hunk 1]
```diff
@@ -1270,7 +1270,7 @@ Test whether `A` is upper triangular starting from the `k`th superdiagonal.
 
 # Examples
 ```jldoctest
-julia> a = [1 2; 2 -1]
+julia> a = Int64[1 2; 2 -1]
```
Comment (KristofferC) [inline stdlib/LinearAlgebra/src/generic.jl:1273]: Why add this?

Comment (jishnub) [reply stdlib/LinearAlgebra/src/generic.jl:1273]: Ok, in that case these aren't necessary

**Comment by fredrikekre** [comment]
The added examples seem to demonstrate the same thing as the first example of respective docstring?

**Comment by jishnub** [comment]
Yes, it's the same thing, but the banded structure made it easier for me to grasp the idea, so I thought of including such examples. Perhaps only one example needs to be added, though, so I'll remove the middle one in each case.

**Comment by jishnub** [comment]
Gentle bump. If this seems reasonable, I would like to update the example, as the new one seems clearer to me

**Comment by jishnub** [comment]
Gentle bump. If there are no objections, this should be good to go?

**Pull request merged**